src/zenml/exceptions.py

```diff
@@ -299,3 +299,7 @@ def __str__(self) -> str:
             the error message
         """
         return f"{self.error}: {self.error_description or ''}"
+
+
+class ReservedNameError(ZenMLBaseException):
+    """Raised when entity is created using reserved name."""
```

Comment:
Not sure who the person to confirm this is (@bcdurak or @AlexejPenner maybe?) but I think we decided at some point to not create these custom ZenML exceptions unless it was really necessary going forward?